src/levanter/data/loader.py

```diff
@@ -110,12 +106,32 @@ def __init__(
             self.scheduler = BatchSchedule(batch_size)
 
         self._batch_sharding = hax.partitioning.sharding_for_axis(self.batch_axis_name, axis_resources, mesh)
+        with mesh:
+            self._data_axis_size = hax.partitioning.physical_axis_size(self.batch_axis_name, axis_resources)
+
+        self._allow_non_divisible_batch_size = allow_nondivisible_batch_size
+        self._pad_final_batch = pad_final_batch
 
         with local_cpu_mesh():
             # It's important that all data loading happens CPU side. We might relax this one day.
-            self._ex_leaves, self._ex_structure = jax.tree_flatten(
-                blocking_wait(self.data_store.getitem_async(0)), is_leaf=is_named_array
-            )
+            current_len = blocking_wait(self.data_store.current_len())
+            if current_len <= 0:
+                logger.warning("Data store currently has no data. We will block until data is available.")
+
+            initial_example = blocking_wait(self.data_store.getitem_async(0))
+            self._ex_leaves, self._ex_structure = jax.tree_flatten(initial_example, is_leaf=is_named_array)
+            self._padding_example = _make_padding_example(initial_example)
```

Comment:
I'm not following this right now - I don't see where `self._padding_example` actually gets used?

Reply:
line 330